OnlineShop.Services.CatalogAPI/Models/Data/Product.cs

```diff
@@ -0,0 +1,18 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace OnlineShop.Services.CatalogAPI.Models.Data
+{
+    public class Product
+    {
+        [BsonId]
+        [BsonRepresentation(BsonType.ObjectId)]
+        public string Id { get; set; }
+        [BsonElement("Name")]
+        public string Name { get; set; }
+        public decimal Price { get; set; }
+        public string Description { get; set; }
+        public string Category { get; set; }
+        public Dictionary<string, string> OptionalFields { get; set; }
+    }
```

Comment:
if its neccessary for you not to have same values in keys then leave it this way otherwise better to create separate model for optional fileds